src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/New-ArmApplicationGatewayResource.ps1

```diff
@@ -17,6 +17,11 @@ function New-ArmApplicationGatewayResource {
         [ValidateSet("Prevention", "Detection")]
         [string]
         $FirewallMode = "Prevention"
+        [bool]
+        $RequestBodyCheck = $false,
```

Comment:
Change type to switch
Rename to "EnableRequestBodyCheck"
With a switch no need to add "=$false"